### PR TITLE
Allow for configurable scheduler fallback period

### PR DESCRIPTION
**Pull request by lowercase00**
This adds a parameter to the Scheduler called
`fallback_period`, which determines the period
before work falls back to a new scheduler. 
Also reduces the default from 600 seconds to 300.
This improves #1758

### CODE REVIEW

File: rq/scheduler.py [hunk 1]
```diff
@@ -98,7 +101,7 @@ def should_reacquire_locks(self):
             return False
         if not self.lock_acquisition_time:
             return True
-        return (datetime.now() - self.lock_acquisition_time).total_seconds() > 600
+        return (datetime.now() - self.lock_acquisition_time).total_seconds() > self.fallback_period
```
Comment (selwin) [inline rq/scheduler.py:104]: I think we can reduce this even further to 3 minutes. What do you think?

File: rq/scheduler.py [hunk 2]
```diff
@@ -36,11 +37,13 @@ class RQScheduler:
 
     def __init__(self, queues, connection, interval=1, logging_level=logging.INFO,
                  date_format=DEFAULT_LOGGING_DATE_FORMAT,
-                 log_format=DEFAULT_LOGGING_FORMAT, serializer=None):
+                 log_format=DEFAULT_LOGGING_FORMAT,
+                 fallback_period=DEFAULT_SCHEDULER_FALLBACK_PERIOD, serializer=None):
```
Comment (selwin) [inline rq/scheduler.py:41]: I don't think we need to expose this parameter as this is a strictly internal thing. After we lower this threshold to 120 seconds, I think we no longer need this parameter, keeping things simpler.

Comment (lowercase00) [reply rq/scheduler.py]: I made a new commit removing the parameter, but keeping the value as a configuration (coming from the `DEFAULT_SCHEDULER_FALLBACK_PERIOD`). That way it should be more flexible to change in the future. Agreed that 120 would be better as well.

**Comment by selwin** [comment]
Thanks!

**Pull request merged**